### PR TITLE
IA-2907 New API endpoint for Org Unit Tree

### PR DESCRIPTION
**Pull request by kemar**
New API endpoint for Org Unit Tree (used by the tree picker in the front-end).

Related elements:

- JIRA ticket : [IA-2907](https://bluesquare.atlassian.net/browse/IA-2907)

## Doc

- [Design doc 1 - Management of org units with an unvalidated parent](https://docs.google.com/document/d/1sa1QHWBlS-K_OitzLzgEOhLGv6nphpAd0z53-ZCak5s)
- [Design doc 2](https://docs.google.com/document/d/1LczWRYbDCJtbxG2fd_c1WtQ4DmNclRG2ixXnr_C58wo)

### Changes to the `Org Unit Tree` API

This PR provides a common endpoint for exploring and searching the Org Unit Tree:

- `/api/orgunits/tree/`
- `/api/orgunits/tree/search/?search=congo&page=2&limit=10`

Params are optional:

| Old  | New | Type | Notes |
| ------------- | ------------- | ------------- | ------------- |
| `rootsForUser`  | *removed*  | | |
| `parent_id`  | `parent_id` | `number` | If not provided, the `root` of the tree is returned  |
|  *N/A* |`force_full_tree`  | `bool` | Force full access to the tree |
| `ignoreEmptyNames`  | `ignore_empty_names`  | `bool` | The case is different |
| `defaultVersion`  | `version`  | `integer` | Version ID |
| `source`  | `data_source_id `  | `integer` | Data Source ID |
| `validation_statuses` | `validation_status`  | `string` | Can be combined |
| *N/A* | `search`  | `string` | Used to search the tree |

### Changes for `DataSource`

Allow to customise the tree display with options on DataSource.

This change is reflected in the API via the `tree_config_status_fields`:

```js
HTTP 200 OK
Allow: GET, PUT, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "id": 2,
    …
    "tree_config_status_fields": [
        "VALID"
    ]
}
```

screen

## How to test

Use DRF UI at `/api/orgunits/tree/`:

screen

[IA-2907]: https://bluesquare.atlassian.net/browse/IA-2907?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: iaso/api/org_unit_tree/views.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        user = self.request.user
+
+        querystring_serializer = OrgUnitTreeQuerystringSerializer(data=self.request.query_params)
+        querystring_serializer.is_valid(raise_exception=True)
+        force_full_tree = querystring_serializer.validated_data.get("force_full_tree")
+        parent_id = querystring_serializer.validated_data.get("parent_id")
+
+        can_view_full_tree = any([user.is_anonymous, user.is_superuser, force_full_tree])
+        if can_view_full_tree:
+            qs = OrgUnit.objects.all()
```
Comment (madewulf) [inline iaso/api/org_unit_tree/views.py]: This seems surprisingly wide (it should be filtered by the account, I would say)

Comment (madewulf) [reply iaso/api/org_unit_tree/views.py]: When a user is anonymous, he should provide a data source

Comment (kemar) [reply iaso/api/org_unit_tree/views.py]: This is fixed in the new code.

**Review by mathvdh — APPROVED** [review]
Nice job !

**Pull request merged**